### ISSUE TITLE
initializer not work excepted

### ISSUE THREAD

**Issue by EvenIsLee** [apache/incubator-mxnet#14490]
when I use @mx.init.register to set a new initializer, bias and others not update excepted. Only weight is changed.

```
import mxnet as mx
from mxnet.gluon import nn

@mx.init.register
class MyInit(mx.init.Initializer):
    def __init__(self, **kwargs):
        super(MyInit, self).__init__(**kwargs)
        #self._verbose_print=True
    def _init_weight(self, name, arr):
        arr[:] = 0.5

    def _init_bias(self, name, arr):
        arr[:] = 1.0

    def _init_gamma(self, name, arr):
        arr[:] = 2.0

    def _init_default(self, name, arr):
        arr[:] = 2.0
   
class Mymodel(nn.Block):
    def __init__(self):
        super(Mymodel, self).__init__()

        with self.name_scope():
            self.dense = nn.Dense(128, in_units=16)
            self.bn = nn.BatchNorm()

    def forward(self, x):
        x = self.dense(x)
        x = self.bn(x)
        return x

if __name__=='__main__':
    data = mx.nd.random.uniform(shape=(1, 16))
    net = Mymodel()
    net.initialize(MyInit())
    net(data)
    print net.dense.weight.data()
    print net.dense.bias.data()
    print net.bn.gamma.data()

```
these are outputs

```
[[0.5 0.5 0.5 ... 0.5 0.5 0.5]
 [0.5 0.5 0.5 ... 0.5 0.5 0.5]
 [0.5 0.5 0.5 ... 0.5 0.5 0.5]
 ...
 [0.5 0.5 0.5 ... 0.5 0.5 0.5]
 [0.5 0.5 0.5 ... 0.5 0.5 0.5]
 [0.5 0.5 0.5 ... 0.5 0.5 0.5]]
<NDArray 128x16 @cpu(0)>

[0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0.
 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0.
 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0.
 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0.
 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0.
 0. 0. 0. 0. 0. 0. 0. 0.]
<NDArray 128 @cpu(0)>

[1. 1. 1. 1. 1. 1. 1. 1. 1. 1. 1. 1. 1. 1. 1. 1. 1. 1. 1. 1. 1. 1. 1. 1.
 1. 1. 1. 1. 1. 1. 1. 1. 1. 1. 1. 1. 1. 1. 1. 1. 1. 1. 1. 1. 1. 1. 1. 1.
 1. 1. 1. 1. 1. 1. 1. 1. 1. 1. 1. 1. 1. 1. 1. 1. 1. 1. 1. 1. 1. 1. 1. 1.
 1. 1. 1. 1. 1. 1. 1. 1. 1. 1. 1. 1. 1. 1. 1. 1. 1. 1. 1. 1. 1. 1. 1. 1.
 1. 1. 1. 1. 1. 1. 1. 1. 1. 1. 1. 1. 1. 1. 1. 1. 1. 1. 1. 1. 1. 1. 1. 1.
 1. 1. 1. 1. 1. 1. 1. 1.]
<NDArray 128 @cpu(0)>
```

**Comment by arcadiaphy** [comment]
@EvenIsLee `net.initialize` only set global default init, which has lowest precedence. The correct way to set init:
```
import mxnet as mx
from mxnet.gluon import nn

class InitWeight(mx.init.Initializer):
    def _init_weight(self, name, arr):
        arr[:] = 0.5

class InitBias(mx.init.Initializer):
    def _init_weight(self, name, arr):
        arr[:] = 1.0


class InitGamma(mx.init.Initializer):
    def _init_weight(self, name, arr):
        arr[:] = 2.0

class Mymodel(nn.Block):
    def __init__(self):
        super(Mymodel, self).__init__()

        with self.name_scope():
            self.dense = nn.Dense(128, in_units=16,
                                  weight_initializer=InitWeight(),
                                  bias_initializer=InitBias())

    def forward(self, x):
        x = self.dense(x)
        return x


if __name__=='__main__':
    data = mx.nd.random.uniform(shape=(1, 16))
    net = Mymodel()
    net.initialize()
    print net.dense.weight.data()
    print net.dense.bias.data()
```

But the implementation in gluon in really confusion compared to that in Module:
```
    >>> # Create and register a custom initializer that
    ... # initializes weights to 0.1 and biases to 1.
    ...
    >>> @mx.init.register
    ... @alias('myinit')
    ... class CustomInit(mx.init.Initializer):
    ...   def __init__(self):
    ...     super(CustomInit, self).__init__()
    ...   def _init_weight(self, _, arr):
    ...     arr[:] = 0.1
    ...   def _init_bias(self, _, arr):
    ...     arr[:] = 1
    ...
    >>> # Module is an instance of 'mxnet.module.Module'
    ...
    >>> module.init_params("custominit")
    >>> # module.init_params("myinit")
    >>> # module.init_params(CustomInit())
```

Sometimes, I want to replace all init in gluon block just like in Module, and expect `Block.initialize` to be just like `Module.init_params`.

**Comment by zachgk** [comment]
Thank you for submitting the issue! I'm labeling it so the MXNet community members can help resolve it.

@mxnet-label-bot add [Python, Gluon]
